src/token-group/generic-token-group.tsx

```diff
@@ -50,19 +53,69 @@ function GenericTokenGroup<Item>(
   const baseProps = getBaseProps(props);
   const className = clsx(baseProps.className, styles.root, hasItems && styles['has-items']);
 
+  useEffect(() => {
```

Comment:
Presumably this is in an effect so that it's delayed until after the surrounding application has updated the state as a result of `onDismiss`? Can we reasonably rely on that? It seems like it's making quite a few assumptions about React internals and the surrounding application.

Reply:
It makes the following assumptions:

1. onDismiss removes the target item and and doesn't add, remove or change position of other items;
2. onDismiss is happening with a state update w/o an extra async delay.

The first assumption is something we don't expect to be violated. The second can be violated e.g. if the update would be wrapped in a `setTimeout`. In that case the focus would remain on the current item and then transition to the next focusable element - same as it was before. If we ever have this situation the only way to fix that would be to expose the token refs. However, the current solution would still work as a reasonable default.

Reply:
ok, I think this makes sense. Might even steal this approach for some applayout focus handling fixes...